contrib/babelfishpg_tds/src/backend/tds/guc.c

```diff
@@ -127,51 +128,57 @@ check_version_number(char **newval, void **extra, GucSource source)
 	len = strlen(*newval);
 	copy_version_number = palloc0(len + 1);
 	memcpy(copy_version_number, *newval, len);
+	current_version = (char *) GetConfigOption("babelfishpg_tds.product_version", true, false);
```

Comment:
1. We should not assign a pointer from `GetConfigOption` (ref: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/BABEL_3_X_DEV__PG_15_X/src/backend/utils/misc/guc.c#L8505-L8506)
2. Since this is also inside `babelfishpg_tds` library, can we use the global variable directly?